src/java.base/share/classes/java/util/Properties.java

```diff
@@ -903,12 +916,23 @@ private void store0(BufferedWriter bw, String comments, boolean escUnicode)
         if (comments != null) {
             writeComments(bw, comments);
         }
-        bw.write("#" + new Date().toString());
-        bw.newLine();
+        writeDateComment(bw);
+
         synchronized (this) {
-            for (Map.Entry<Object, Object> e : entrySet()) {
-                String key = (String)e.getKey();
-                String val = (String)e.getValue();
+            @SuppressWarnings("unchecked")
+            Collection<Map.Entry<String, String>> entries = (Set<Map.Entry<String, String>>) (Set) entrySet();
+            // entrySet() can be overridden by subclasses. Here we check to see if the returned instance type is the one
+            // returned by the Properties.entrySet() implementation. If yes, then we sort those entries
+            // in the natural order of their key. Else, we consider that the subclassed implementation may potentially
+            // have returned a differently ordered entries and so we just use the iteration order of the returned instance.
+            if (entries instanceof Collections.SynchronizedSet<?> ss
```

Comment:
Would it be possible to re-format the comment to keep the lines a bit inconsistent, it's just a bit annoying to have overly long lines just here.

Reply:
Done. I've updated the PR to reduce the line length of that code comment.